qiskit_nature/properties/second_quantization/electronic/electronic_driver_result.py

```diff
@@ -0,0 +1,103 @@
+from qiskit_nature.drivers.second_quantization import QMolecule
+from qiskit_nature.operators.second_quantization import FermionicOp
+
+from ..driver_result import DriverResult
+from ..second_quantized_property import LegacyDriverResult, LegacyElectronicDriverResult
+from .angular_momentum import AngularMomentum
+from .bases import ElectronicBasis, ElectronicBasisTransform
+from .dipole_moment import TotalDipoleMoment
+from .electronic_energy import ElectronicEnergy
+from .magnetization import Magnetization
+from .particle_number import ParticleNumber
+
+
+class ElectronicDriverResult(DriverResult):
+    """The ElectronicDriverResult class.
+
+    This is a :class:~qiskit_nature.properties.CompositeProperty gathering all property objects
+    previously stored in Qiskit Nature's `QMolecule` object.
+    """
+
+    def __init__(self) -> None:
+        """
+        Property objects should be added via `add_property` rather than via the initializer.
+        """
+        super().__init__(self.__class__.__name__)
+        self.molecule: Molecule = None
+        self.electronic_basis_transform: ElectronicBasisTransform = None
+        # TODO: add origin driver metadata
+        # TODO: where to put orbital_energies?
+        # TODO: where to put kinetic, overlap matrices? Do we want explicit Fock matrix?
+
+    @classmethod
+    def from_legacy_driver_result(cls, result: LegacyDriverResult) -> "ElectronicDriverResult":
+        """Converts a QMolecule into an `ElectronicDriverResult`.
+
+        Args:
+            result: the QMolecule to convert.
+
+        Returns:
+            An instance of this property.
+
+        Raises:
+            QiskitNatureError: if a WatsonHamiltonian is provided.
+        """
+        cls._validate_input_type(result, LegacyElectronicDriverResult)
+
+        ret = cls()
+
+        qmol = cast(QMolecule, result)
+
+        ret.add_property(ElectronicEnergy.from_legacy_driver_result(qmol))
+        ret.add_property(ParticleNumber.from_legacy_driver_result(qmol))
+        ret.add_property(AngularMomentum.from_legacy_driver_result(qmol))
+        ret.add_property(Magnetization.from_legacy_driver_result(qmol))
+        ret.add_property(TotalDipoleMoment.from_legacy_driver_result(qmol))
+
+        ret.electronic_basis_transform = ElectronicBasisTransform(
+            ElectronicBasis.AO, ElectronicBasis.MO, qmol.mo_coeff, qmol.mo_coeff_b
+        )
+
+        geometry: List[Tuple[str, List[float]]] = []
+        for atom, xyz in zip(qmol.atom_symbol, qmol.atom_xyz):
+            geometry.append((atom, xyz))
+
+        ret.molecule = Molecule(geometry, qmol.multiplicity, qmol.molecular_charge)
+
+        return ret
+
+    def second_q_ops(self) -> List[FermionicOp]:
+        """Returns the list of `FermioncOp`s given by the properties contained in this one."""
+        ops: List[FermionicOp] = []
+        # TODO: make aux_ops a Dict? Then we don't need to hard-code the order of these properties.
```

Comment:
What is the status of making the `aux_operators` a `Dict` in the entire stack? Would this be in-scope for the 0.2.0 release?

Reply:
Ah I see I just commented about this above in an interpret method seeing a bunch of hard-coded constants. Clearly this needs changing in Terra to go all the way through; do not know if it can be changed in Nature and adapted to a list in the interim of having it all the way.

Reply:
I had a quick look at Terra and [here](https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/algorithms/minimum_eigen_solvers/minimum_eigen_solver.py#L33) the `MinimumEigensolver` interface already enforces this to be a list. (An analogue case exists in the `Eigensolver` interface.)

I think, for the time we should get this PR done as is but we should open an issue about this in Terra.

If you want to, we can use an `OrderedDict` internally in the meantime. However, we will have to be careful when dealing with aux operators getting removed e.g. due to symmetry constraints, etc.

Reply:
Yes the (Minimum)Eigensolver interfaces were never changed away from the List they always were. The removal of operators due to symmetry constraints is entirely at this level. Terra will act on whatever is in the list and give back results - any adaptation at this level, ahead of it being supported by Terra, would be around the call to the (Minimum)EigenSolver and its result, which would still be a list. It works as a list for now - its something thats going to affect the VQEProgram too - so maybe this is best left as-is for the time being though I think we all agree its an area that needs to be improved on.

Reply:
https://github.com/Qiskit/qiskit-terra/issues/6772